primitives/linux-port/column.cpp

```diff
@@ -895,6 +976,75 @@ inline bool nextColValue(
   return true;
 }
 
+template <typename T, int COL_WIDTH>
```

Comment:
Could you elaborate why did you make this nextColValueAux that is a very similar copy of nextColValue?

Reply:
This is to avoid the `hasAuxCol` based if branching inside the for loop in `scalarFiltering()`:

```c++
  auto nextColValuePtr = hasAuxCol ? nextColValueAux<T, WIDTH> : nextColValue<T, WIDTH>;                                                    
  
  // Loop over the column values, storing those matching the filter, and updating the min..max range                                        
  for (uint32_t i = initialRID; (*nextColValuePtr)(curValue, &isEmpty, &i, &rid, srcArray, srcSize,                                         
                                                   ridArray, ridSize, outputType, emptyValue,                                               
                                                   blockAux, emptyValueAux);)                                                               
  {
    // loop code
  }
```

Reply:
You can do it a bit shorter like [this](https://godbolt.org/z/K8xW19q89). The example isn't complete but consider if constexpr usage.

Reply:
Thanks for showing the usage of `if constexpr`, I was not aware of this. I have made this change.